demo.py

```diff
@@ -0,0 +1,146 @@
+class Resource(models.Model):
+    client = ForeignKey(Client)
+    owned_by = ForgeignKey(Resource, null=True)
+    name = CharField(blank=False, max_length=128)
+    configuration = JSONField(null=False)
```

Comment:
How does this configuration field matches/interacts with the existing configuration in OpenCraft IM? Ie:
* The django config set from the env
* The configuration set in model fields on OpenEdXInstance 
* The configuration set in ansible/yaml (from the PR for example)

I assume that Resources are meant to be subclassed, right? Why not keeping it as individual model attributes?